linked_list/implementation.py

```diff
@@ -7,34 +8,122 @@ class LinkedList(AbstractLinkedList):
+        
     def __str__(self):
-        pass
+        return '{}'.format([i for i in self])
 
     def __len__(self):
-        pass
+        count = 0
+        for i in self:
+            count += 1
+        return count
 
     def __iter__(self):
-        pass
+        if self.start: # if the LinkedList object has some value
+            current = self.start # assign the self.start node to our current working node
+            if current.next: # if current has a 'next' pointer to another node (i.e. there is more than 1 element in the input)
+                nxt = self.start.next # assignt its 'next' node to our nxt working node
+            yield current.elem # return the element of the current working node, then pause
+            while current.next: # if the current working node had a 'next' pointer
+                yield current.next.elem # return the element of the 'next' pointer of the current working node
+                current = nxt # we finished with the current node, let's assign the new current working node to be the next one
+                nxt = current.next # and let's assign the NEW current working node's 'next' pointer to our nxt working variable
+        
 
     def __getitem__(self, index):
-        pass
+        length = len(self)
+        if index < 0 and abs(index) > length:
+            raise IndexError
+        if length == 0 or length <= index:
+            raise IndexError
+        if index >= 0:
+            for ind,element in enumerate(self):
+                if ind == index:
+                    return element                  
+        else: #negative indexing
+            for ind,element in enumerate(self):
+                if ind - length == index: #defining relationship between negative index and positive index
+                    return element
+            # for idx, element in self.elements: 
+                # no idea how but gotta get the index + element somehow
+                # if idx == index:
+                    # return element
 
     def __add__(self, other):
-        pass
+        copy = LinkedList(self) # created copy so we don't overwrite anything
+        for i in other:
+            copy.append(i) # append function is niiiceeeee
+        return copy
 
-    def __iadd__(self, other):
-        pass
+    def __iadd__(self, other): # += operation, same as __add__ except we do want to overwrite the original LL
+        for i in other:
+            self.append(i)
+        return self
 
     def __eq__(self, other):
-        pass
+        if len(self) == 0 and len(other) == 0: # if we have two empty LL's
+            return True
+        if not (self and other): #if either aren't filled
+            return False
+        if len(self) != len(other): #if lengths aren't equal
+            return False
+        for i,j in zip(self, other): # create a zip structure to compare parallel lists
```

Comment:
Sebastian wrote this part when I was asleep so I gotta check it out too. XD thanks for highlighting it.